### DISCUSSION TITLE
How to capture error from auth0 rule?

### DISCUSSION THREAD

**Discussion by shahkeyur** [nextauthjs/next-auth#9263] [Help]
I have some auth0 rules which reject authentication based on some criteria. For e.g. if email is not verified. 

How do I capture such case from next-auth without repeating same condition on server. I want to keep the flow of auth0 rules, just redirect to different page if there was an error by reading it's error code.

**Comment by Walim-Akki** [comment]
Hello

Did you find a solution ? 
I face the same issue.

I have the following auth0 action : 

exports.onExecutePostLogin = async (event, api) => {
  if (!event.user.email_verified) {
    api.access.deny('email_not_verified');
  }
};


I am  trying to catch the error on my front page, i have this error on my console : 

[next-auth][error][OAUTH_CALLBACK_HANDLER_ERROR] 
https://next-auth.js.org/errors#oauth_callback_handler_error access_denied {
  error: {
    message: 'access_denied',
    stack: 'Error: access_denied\n' +
      '    at oAuthCallback (/app/node_modules/next-auth/core/lib/oauth/callback.js:39:19)\n' +
      '    at Object.callback (/app/node_modules/next-auth/core/routes/callback.js:52:39)\n' +
      '    at AuthHandler (/app/node_modules/next-auth/core/index.js:201:41)\n' +
      '    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)\n' +
      '    at <anonymous> (/app/node_modules/@sidebase/nuxt-auth/dist/runtime/server/services/authjs/nuxtAuthHandler.mjs:88:24)\n' +
      '    at async Object.handler (file:///app/node_modules/h3/dist/index.mjs:1630:19)\n' +
      '    at async Server.toNodeHandle (file:///app/node_modules/h3/dist/index.mjs:1840:7)',
    name: 'Error'
  },
  error_description: 'email_not_verified',
  providerId: 'xxxxxxxxxx',
  message: 'access_denied'
}

I did not find a way to catch this error
